### ISSUE TITLE
Crash in CResource::Load

### ISSUE THREAD

**Issue by Dutchman101** [multitheftauto/mtasa-blue#1077]
**Describe the bug**
There is a crash on client.dll on an assert located at [https://github.com/multitheftauto/mtasa-blue/blob/4244ec52a5155d624cd5ec8dca1bdca067309358/Client/mods/deathmatch/logic/CResource.cpp#L345](https://github.com/multitheftauto/mtasa-blue/blob/4244ec52a5155d624cd5ec8dca1bdca067309358/Client/mods/deathmatch/logic/CResource.cpp#L345).

EAX is 00000001 and the question is: in which scenario is it possible to not have a resource root entity, as a result of which we hit the assert?

Example dump trace: [https://pastebin.com/ud5bFj6L](https://pastebin.com/ud5bFj6L)

**To reproduce**
No known reproduction. This is a popular crash on client.dll (top 10).

**Expected behaviour**
Either avoid having no resource root entity (by finding out where nullptr comes from), or don't let this condition be fatal.

**Version**
1.5.6-r18938

**Additional context**

```
CResource::CResource(..., CClientEntity* pResourceEntity, ...)
    m_pResourceEntity = pResourceEntity;
```
CResource doesn't create the object
It only keeps a pointer to it

Pointer comes from somewhere else
CPacketHandler::Packet_ResourceStart(NetBitStreamInterface& bitStream) calls ->Add

```
// Get the resource entity
    CClientEntity* pResourceEntity = CElementIDs::GetElement(ResourceEntityID);
```

There is no code to check pResourceEntity for NULL

Example crash dumps (2):
[dump.zip](https://github.com/multitheftauto/mtasa-blue/files/3553042/dump.zip)

**Comment by Dutchman101** [comment]
Just discovered that 1 of the 2 attached dumps contains a different (probably more complete) stack trace than the other one which is contained in the pastebin link.

Here:
```
STACK_TEXT:  
0028f358 1a6d6bda eccd3b96 00000000 1d537e30 client!CResource::Load+0x606
0028f434 1a6daecb 0028f914 eccd373e 00000000 client!CPacketHandler::Packet_ResourceStart+0x6ba
0028f89c 1a665de0 0191c352 0028f914 0028f8d0 client!CPacketHandler::ProcessPacket+0xc9b
0028f8ac 04a3619f 0191c352 0028f914 01a67ed0 client!CClientGame::StaticProcessPacket+0x50
WARNING: Stack unwind information not available. Following frames may be wrong.
0028f8d0 04a35155 00000052 0028f914 2efd168f netc!InitNetRev+0x616f
0028f994 1a65d212 0ccef160 0ccef070 0ccef001 netc!InitNetRev+0x5125
0028f9a8 1a65d7f1 00000001 eccd34aa 00436881 client!CClientGame::DoPulses2+0x82
0028fb08 1a65ceb4 eccd3422 0191c3c8 0199c230 client!CClientGame::DoPulses+0x511
0028fb80 0f5f93c3 0191c3c8 064fae20 0f5b31a5 client!CClientGame::DoPulsePostFrame+0x5f4
0028fb8c 0f5b31a5 a227ef96 065666c0 064fae20 core!CModManager::DoPulsePostFrame+0x53
0028fc7c 0f64e234 a227ef2a 065666c0 00000000 core!CCore::DoPostFramePulse+0x7a5
0028fcc0 0f654c90 064fae20 04148410 007f99b0 core!CDirect3DEvents9::OnPresent+0x164
0028fd24 007f9b12 065666c0 00000000 00000000 core!CProxyDirect3DDevice9::Present+0x30
0028fd98 5b6048f1 00619b71 00000001 00619b71 gta_sa+0x3f9b12
0028fd9c 00619b71 00000001 00619b71 0000001a multiplayer_sa!HOOK_CallIdle+0x11
0028fda0 00000000 00619b71 0000001a 00000001 gta_sa+0x219b71
```

Also starting to get the idea that mostly CCDPlanet and derivative servers (from its leak) have its players face this crash, in specific both of the dumps, independently from server connected to while crashing, containing `car_system` in szResourceName. This would come back to shared usage across CCD-based servers.

Therefore it means that perhaps a resource (like this one) is capable of doing something that leads to the absence of a resource root entity/null, something which can still never be the intended behavior.

**Comment by botder** [comment]
Fixed in commit 4d64de66b07fb1b4fd0f9cf551bd9f88ea82130a. Server sent an invalid element id for the starting client resource.
